### PR TITLE
Feature/cc 2711 3

### PR DESCRIPTION
**Pull request by rDr4g0n**
https://jira.zenoss.com/browse/CC-2711

Makes host status icon not only show if a host is active, but also if it is authenticated.

unauthedhost

With the additional of auth and host keys, it is easy to have a host that is connected, but cannot operate because its auth token is invalid. This makes it more obvious if that is the case, so that the user can resolve the issue.

### CODE REVIEW

File: auth/hostregistry.go [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	// exist in the host expiration registry
+	ErrMissingHost = errors.New("Host is not present in host expiration registry")
+)
+
+// HostExpirationRegistry is a threadsafe map of
+// host id to auth expiration time. NOTE: expired hosts
+// are not removed from the registry
+type HostExpirationRegistry struct {
+	registry map[string]int64
+	sync.Mutex
```
Comment (iancmcc) [inline auth/hostregistry.go]: Please use a `sync.RWMutex` instead. It'll add significant performance benefits in the `IsExpired` call.

File: auth/hostregistry_test.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	. "gopkg.in/check.v1"
+	"time"
+)
+
+func (s *TestAuthSuite) TestHostExpired(c *C) {
+	reg := auth.NewHostExpirationRegistry()
+	hostid := "fakehost"
+	expires := jwt.TimeFunc().Add(time.Minute).Unix()
+	reg.Set(hostid, expires)
+
```
Comment (iancmcc) [inline auth/hostregistry_test.go]: Also add a test here for whether a nonexistent host returns expired

File: facade/host.go [hunk 3]
```diff
@@ -351,6 +363,46 @@ func (f *Facade) FindReadHostsInPool(ctx datastore.Context, poolID string) ([]ho
...
+		}
+
+		status := host.HostStatus{HostID: id, MemoryUsage: service.Usage{}}
+		active, err := f.zzk.IsHostActive(h.PoolID, h.ID)
+		if err != nil {
+			continue
+		}
+		status.Active = active
+
+		authed, _ := f.hostRegistry.IsExpired(h.ID)
```
Comment (iancmcc) [inline facade/host.go]: This seems like a peculiarly named variable

File: web/ui/src/common/resourcesFactory.js [hunk 4]
```diff
@@ -69,6 +69,12 @@
                 method: GET,
                 url: id => `/api/v2/pools/${id}/hosts`
             },
+            // TODO - MOVE THIS WHEN THE V2 API
```
Comment (iancmcc) [inline web/ui/src/common/resourcesFactory.js]: maybe open a ticket for this?

File: facade/host.go (hunk 3 above)
Comment (rDr4g0n) [inline facade/host.go]: I made a separate PR for this egregious error https://github.com/control-center/serviced/pull/3080

**Pull request merged**